docs/extensions/getting-started/github-docs.md

```diff
@@ -0,0 +1,277 @@
+```
+
+### Page metadata
+
+Every extension documentation page requires metadata that tells MakeCode which extension needs to be loaded in order to render the code snippets on the page. At the minimum, this metadata should contain an entry for your own extension in order for any of the snippets to render properly. Extensions are declared in a `package` specification block, typically at the end of the markdown page.
+
+The package metadata follows this format:
+
+````markdown
+```package
+my-custom-extension=github:username/extension-repo
+```
+````
+
+Each entry in the `package` code block should be formatted like so:
+
+```
+<extension-name>=github:<github-username>/<github-repo>
+```
+
+The `<extension-name>` in the above snippet should be the name that appears in the `pxt.json` name field of your extension. It's best practice to make sure that this name always matches the `name` set in your extension's `pxt.json` file.
+
+If your repository contains multiple extensions, add the path to the directory containing the desired extension's `pxt.json` file after the github repo like so:
+
+```
+<extension-name>=github:<github-username>/<github-repo>/path/to/extension
+```
+
+To add more extensions, simply add each on a new line within the `package` code block. This can be useful if your example code needs to reference APIs from another extension:
+
+````markdown
+```package
+my-custom-extension=github:username/extension-repo
+my-other-extension=github:username/different-extension-repo
+```
+````
+
+### Full example markdown
+
+Here is a full example markdown page for a file read API:
+
+````markdown
+# file read
+
+Reads the contents of a file from the file system
+
+```sig
+files.read("/path/to/file.txt", "UTF-8")
+```
+
+## Parameters
+
+* **path**: a string that contains the path to the file on disk
+* **encoding**: the encoding of the file to be read
+
+## Example
+
+This example reads from a file on disk and prints the contents to the console.
+
+```blocks
+const contents = files.read("/path/to/file.txt", "UTF-8");
+console.log(contents)
+```
+
+## See also
+* [write file](./write-file)
+
+```package
+my-custom-extension=github:username/extension-repo
+```
+````
+
+## Step 3: Add the markdown file to pxt.json
+
+All documentation files must be included in the `files` entry of the extension's `pxt.json` file. For example:
+
+```json
+{
+    "name": "my-custom-extension",
+    "version": "0.0.1",
+    "description": "",
+    "dependencies": {
+        "device": "*"
+    },
+    "files": [
+        "main.ts",
+        "README.md",
+        "docs/my-custom-block.md",
+    ],
+    "testFiles": [
+        "test.ts"
+    ],
+    "supportedTargets": [
+        "arcade"
+    ],
+    "preferredEditor": "tsprj"
+}
+
+```
+
+## Step 4: Add a comment annotation to the function
+
+Once you have a markdown file added to your extension, it's time to link that page to the function using a comment annotation by specifying a "help path".
```

Comment:
Is "the function" clear enough? When reading this, I had to do a bit of a double take to make the connection that "the function" is the block definition, but being that I don't make custom blocks very often, it might be clear enough.

Reply:
Or maybe in the example provided the annotation can be above the custom block function signature?

Reply:
Good idea, how's this....